### DISCUSSION TITLE
Get Topics/Tags with published Posts only

### DISCUSSION THREAD

**Discussion by MimisK13** [austintoddj/canvas#1021] [Troubleshooting] [answered]
Hi

How can I get only the published posts by topic or tag? 

```php
$tags = Tag::with(['posts'])->firstWhere('slug', $slug);
```
This also displays the drafts.

Any suggestions? 

Thnx

**Comment by MimisK13** [comment, marked as answer]
And as soon as I asked for help, I thought the following and it worked: 

```php
$tags = Tag::with(['posts' => function($q){
    $q->published();
}])->firstWhere('slug', $slug);
```

**Comment by austintoddj** [comment]
Closing as answered.

On current Canvas, filter to the public snapshot (published, not drafts/pending) in your query/scopes when loading related posts for a tag or topic. v7 still expects host readers to use **live** post columns only — never `pending`.

If you need a concrete v7 query example, open a new discussion against current `main`.
